app/src/routes/api/v2/area.router.js

```diff
@@ -252,19 +216,57 @@ class AreaRouterV2 {
             deforestationAlerts: deforAlertSub,
             webhookUrl,
             monthlySummary: summarySub,
-            subscriptionId: subId,
-            language: lang,
+            language: ctx.request.body.language,
             email
         }).save();
-        ctx.body = AreaSerializerV2.serialize(area);
-    }
 
-    static async save(ctx) {
-        await AreaRouterV2.saveArea(ctx, ctx.state.loggedUser.id);
+        // If no datasets to register, no need to create a subscription
+        if (area.fireAlerts || area.deforestationAlerts || area.monthlySummary) {
+            const subscriptionId = await SubscriptionService.createSubscriptionFromArea(area);
+            if (subscriptionId) {
+                // Update the subscription id in the area and save again
+                area.subscriptionId = subscriptionId;
+                area = await area.save();
+            }
+        }
+
+        ctx.body = AreaSerializerV2.serialize(area);
     }
 
     static async update(ctx) {
-        const area = await AreaModel.findById(ctx.params.id);
+        let previousArea = await AreaModel.findById(ctx.params.id);
+        let area = await AreaModel.findById(ctx.params.id);
+        if (!area) {
+            // Try to find subscription with same ID
+            const [subscription] = await SubscriptionService.findByIds([ctx.params.id]);
+            if (!subscription) {
+                ctx.throw(404, 'Area not found');
+                return;
+            }
+
+            // Create a new area from the subscription
+            area = await new AreaModel({
+                _id: subscription.id,
+                name: subscription.attributes.name,
+                application: subscription.attributes.attributes || 'gfw',
```

Comment:
if this default is indeed needed, tests should fail when I remove it, and they dont

Reply:
Removed the default here because there is one in the Areas model, and that one, when removed, makes the appropriate tests fail :)